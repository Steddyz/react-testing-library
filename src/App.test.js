// import { render, screen, fireEvent } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";

// import App from "./App";

// describe("Test App", () => {
//   test("renders learn react link", () => {
//     render(<App />);
//     const helloWorldElement = screen.getByText(/hello world/i);
//     const btn = screen.getByRole("button");
//     const input = screen.getByPlaceholderText(/input value/i);

//     expect(helloWorldElement).toBeInTheDocument();
//     expect(btn).toBeInTheDocument();
//     expect(input).toMatchSnapshot();
//   });
//   test("renders learn react link", async () => {
//     render(<App />);
//     // const helloWorldElement = screen.queryByText(/hello5252/i);
//     // expect(helloWorldElement).toBeNull();
//     screen.debug();
//     const helloWorldElem = await screen.findByText(/data/i);
//     expect(helloWorldElem).toBeInTheDocument();
//     expect(helloWorldElem).toHaveStyle({ color: "red" });
//     screen.debug();
//   });
//   test("Button Event", () => {
//     render(<App />);

//     const btn = screen.getByTestId("toggle-btn");
//     expect(screen.queryByTestId("toggle-element")).toBeNull();
//     fireEvent.click(btn);
//     expect(screen.queryByTestId("toggle-element")).toBeInTheDocument();
//     fireEvent.click(btn);
//     expect(screen.queryByTestId("toggle-element")).toBeNull();
//   });
//   test("Input Event", () => {
//     render(<App />);

//     const input = screen.getByPlaceholderText(/input value/i);
//     expect(screen.queryByTestId("value-element")).toContainHTML("");
//     // Искуственное событие
//     // fireEvent.input(input, { target: { value: "123123" } });

//     // Воспроизводит действия пользователя
//     userEvent.type(input, "123123");

//     expect(screen.queryByTestId("value-element")).toContainHTML("123123");
//   });
// });
