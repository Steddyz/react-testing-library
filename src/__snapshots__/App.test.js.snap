// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App renders learn react link 1`] = `
<input
  placeholder="input value"
/>
`;

exports[`renders learn react link 1`] = `
<input
  placeholder="input value"
/>
`;
